working_villagers/jobs/brush_collector.lua

```diff
@@ -0,0 +1,131 @@
+local func = working_villages.require("jobs/util")
+
+local shrubs = {
+  -- more priority definitions
+	names = {
+		["default:blueberry_bush_leaves_with_berries"]={},
+		["default:cactus"]={collect_only_top=true},
+		["default:papyrus"]={collect_only_top=true},
+		["default:bush_leaves"]={},
+		["default:bush_stem"]={},
+		--["flowers:mushroom_red"]={},
+	},
+  -- less priority definitions
+	groups = {
+		["leaves"]={},
+		--["wood"]={},
+	},
+}
+
+function shrubs.get_shrub(item_name)
+  -- check more priority definitions
+	for key, value in pairs(shrubs.names) do
+		if item_name==key then
+			return value
+		end
+	end
+  -- check less priority definitions
+	for key, value in pairs(shrubs.groups) do
+		if minetest.get_item_group(item_name, key) > 0 then
+			return value;
+		end
+	end
+	return nil
+end
+
+function shrubs.is_shrub(item_name)
+  local data = shrubs.get_shrub(item_name);
+  if (not data) then
+    return false;
+  end
+  return true;
+end
+
+local function find_herb_node(self)
+	return function(pos)
+		if minetest.is_protected(pos, self:get_player_name()) then return false end
+		if working_villages.failed_pos_test(pos) then return false end
+
+		local node = minetest.get_node(pos);
+  		local data = shrubs.get_shrub(node.name);
+  		if (not data) then
+    			return false;
+  		end
+
+  		if data.collect_only_top then
+    			-- prevent to collect plat part, which can continue to grow
+    			local pos_below = {x=pos.x, y=pos.y-1, z=pos.z}
+    			local node_below = minetest.get_node(pos_below);
+    			if (node_below.name~=node.name) then
+      				return false;
+    			end
+    			local pos_above = {x=pos.x, y=pos.y+1, z=pos.z}
+    			local node_above = minetest.get_node(pos_above);
+    			if (node_above.name==node.name) then
+      				return false;
+    			end
+  		end
+
+  		return true;
+	end
+end
+
+local searching_range = {x = 10, y = 5, z = 10}
+
+local function put_func()
```

Comment:
In manipulate_chest, there's a check: `if (not stack:is_empty()) and (put_func...` This could be modified to check whether type(put_func) == "boolean".

Alternatively, we could add a util.put_all() that simply returns true.